lib/decompress/zstd_decompress_block.c

```diff
@@ -121,8 +123,28 @@ size_t ZSTD_decodeLiteralsBlock(ZSTD_DCtx* dctx,
                     litCSize = (lhc >> 22) + ((size_t)istart[4] << 10);
                     break;
                 }
+                RETURN_ERROR_IF(litSize > 0 && dst == NULL, dstSize_tooSmall, "NULL not handled");
```

Comment:
Question : 
is this scenario _possible_,
for example, could it happen if the user would provide a `NULL` pointer as destination,

or should it normally never happen, 
for example because the code calling this function will ensure it never provides `NULL` as `dst` ?

If this is the second case, this condition is rather an `assert()`,
and it should also be clearly documented, so that the caller knows that this scenario is not allowed.

It seems like a small detail (is it an error ? or is it forbidden ?), but this kind of decision trickles down throughout the whole code, so it's important to be familiar with the logic.

Reply:
there's a fuzzer test that passes NULL and expects this error response, this was added to address it

Reply:
nit: I would re-write this check as `litSize > dstCapacity`, which is more general.

Edit: The check below `expectedWriteSize < litSize` should cover this case.